### ISSUE TITLE
How to maximise importer performance ?

### ISSUE THREAD

**Issue by rushilagr** [jbarrasa/neosemantics#28]
I have access to a high performance machine, 32 core, 128GB ram.
Any tips on how to maximise performance?

Are there any heuristics I could use to set the commitSize and the nodeCacheSize?
Would allocating more memory to the JVM running the neo4j server process help?

I am importing the entire DBpedia dataset (144gb) so speedy imports would really help.

Thanks!

**Comment by jbarrasa** [comment]
Hi, sorry for the late reply. 
I have to say I've only tried it on my 4 core 16G ram machine so would be super keen to hear about your experience if you could share it.
My first suggestion would be to probably try with a large nodeCacheSize to use as much memory as possible and maybe work from there. 
FYI, the biggest dataset I've tried neosemantics on is the British Library catalogue, which is ~130 million triples.

Looking forward to hearing about your experience!

JB.

**Comment by rushilagr** [comment]
Hi JB,

First of all, thanks a lot of writing this importer! It worked beautifully and made my life a hell of a lot easier.

These are some insights/details -
- Total ttl file size was 90gb uncompressed. Post import DB size was ~90GB as well.
- Machine specs - 4 core - 50gb ram - SSD machine on GoogleCloud

- The most effective technique to speed up import was to have enough RAM available to hold the entire DB in memory - 90gb in my case. Towards the end, the import process slowed down by a magnitude of 100 when the RAM utilisation maxed out (due to the importer having to hit the harddisk for node lookup, I assume). When I allocated more RAM to about 90gb, the entire process happened quite smoothly. 

- Import speed averaged around 10GB / hour. Peaking at the begin at 13gb/hour and gradually slowing down.

- Some surpirsing behaviour - towards the end of the process, the DB size had grown to about 120gb and then gradually shrank to 90gb at the end. After some usage, the DB has considerably shrunk to around 30GB! 
 
So yeah, thanks !

**Comment by jbarrasa** [comment]
Thanks for sharing! This is great to hear :) 
So was the DBpedia dataset that you imported?

I'd be curious to know if possible a few more details if you have them at hand. Here are some ideas: 
number of triples in the original RDF 
number of nodes in the Neo4j graph 
any interesting queries or results that you can share?
version of Neo4j that you used?

(Feel free to take this conversation off-line if you prefer)

Thanks a lot and looking forward to improving neosemantics and making it useful for more people.

JB.

**Comment by rushilagr** [comment]
Would like to take the conversation offline. What do you prefer?
If you prefer email then - rushil.agrawal@gmail.com

Thanks :)

**Comment by giantamoeba** [comment]
I am also interested in this point, and would like to thank you for this awesome importer!

My files are not that big (50 MB) but they come frequently and need to be handled as fast as possible. I have achieved in my docker container to load 700k triples from two files in ca 37 seconds after disabling `verifyUriSyntax` and setting `nodeCacheSize` to 100k.

I have tried to parallelize the loading using two threads in python, but to my surprise this did not have any effect on the loading time whatsoever! As if the two files are being loaded one after the other
